openmmtools/multistate/multistatesampler.py

```diff
@@ -950,6 +978,13 @@ def _read_options(check_iteration):
         self._last_mbar_f_k = last_mbar_f_k
         self._last_err_free_energy = last_err_free_energy
 
+        # Handle with context caches as specified
+        # update context caches attributes handling inputs
+        self.energy_context_cache, self.sampler_context_cache = self._initialize_context_caches(
```

Comment:
I think my only comment is that we discussed this could be simplified into a single `self._initialize_context_cache()` that just returns the global context cache, and there is no need to accept the kwargs for context caches since we can just set them as public attributes now, but I don't feel very strongly about this either way. I think it's just simpler and more streamlined if we don't have to pass these through various ways of creating or initializing the class, but it's not too critical either way.

Reply:
I think I understand what you mean. For the sake of this release I think we are good with having this as it is now, but I'll raise a new issue where we can discuss enhancing this behavior.

Reply:
Did you raise the issue? If you did can you link to John's comment?